src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java

```diff
@@ -100,24 +89,35 @@ private void processEnvelopes(String containerName, List<Envelope> envelopes)
         }
     }
 
-    private void processEnvelope(CloudBlobContainer container, Envelope envelope)
-        throws IOException, StorageException, URISyntaxException {
+    private void processEnvelope(String containerName, ContainerURL container, Envelope envelope)
+        throws IOException {
 
-        log.info("Processing zip file {}", envelope.getZipFileName());
+        String zipFileName = envelope.getZipFileName();
+        log.info("Processing zip file {}", zipFileName);
 
-        CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(envelope.getZipFileName());
-        BlobInputStream blobInputStream = cloudBlockBlob.openInputStream();
+        BlockBlobURL blockBlobURL = container.createBlockBlobURL(zipFileName);
 
-        try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
-            ZipFileProcessor zipFileProcessor = processZipInputStream(zis, envelope);
+        BlobGetPropertiesResponse properties = blockBlobURL
+            .getProperties(new BlobAccessConditions(), Context.NONE)
+            .blockingGet();
 
-            if (zipFileProcessor != null) {
-                processParsedEnvelopeDocuments(
-                    envelope,
-                    zipFileProcessor.getPdfs(),
-                    cloudBlockBlob,
-                    serviceBusHelper
-                );
+        boolean leaseAvailable = azureStorageHelper
+            .checkLeaseAvailable(containerName, zipFileName, properties.headers().leaseState());
+
+        if (leaseAvailable) {
```

Comment:
a .filter here could achieve the same thing ... then a .switchIfEmpty to log the error.